### PR TITLE
Adding Dashboard with placeholder widgets

### PR DESCRIPTION
**Pull request by Durbatuluk1701**
Not yet ready to merge, just opening for viewing

### CODE REVIEW

**Comment by maxxxxxdlp** [comment]
Awesome. In that case, I converted it to a `Draft Pull Request`. You can convert it back once it is ready

**Comment by Durbatuluk1701** [comment]
Couple things to note:
- I think we should add a scrollbar to the dashboard, probably a separate story (either that or we should workshop how we want the dashboard to look a bit more)
- I do not quite understand the Tailwind style for styling element, probably could be refactored to work into Tailwind soon. 
- Seems to be an issue with the "Create" button for google calendar rendering over the Dashboard, need to look more into how the Portal works and why it cannot show up above it

**Comment by maxxxxxdlp** [comment]
Do you want to look into [Tailwind documentation](https://tailwindcss.com/docs/editor-setup) and refactor the code to use tailwind class names, or do you want me to refactor it?

**Comment by Durbatuluk1701** [comment]
I will refactor for Tailwind as long as this PR isn't massively blocking others, just will take some time to read the docs and refactor.

**Comment by Durbatuluk1701** [comment]
Ready for review and discussion. Feel like there may be some more work to tweak the widgets to look more how we want, but until I have a better idea of how we want to format it I am not really sure

File: src/src/declarations.d.ts [hunk 1]
```diff
@@ -17,6 +17,13 @@ import type { IR, RA, RR } from './utils/types';
 /* eslint-disable functional/prefer-readonly-type */
 
 declare global {
+
```
Comment (maxxxxxdlp) [inline src/src/declarations.d.ts:20]: This file is only for changing the type definitions of default javascript methods to make them more type safe.

Your custom typings should probably go into a different file. `./index.tsx` or `./Dashhboard/index.tsx` would be a good place for this at the moment.

Though, if you **really** want the type to be in global scope, then create a separate `.d.ts` file and put `WidgetObj` there

File: src/src/components/App/index.tsx [hunk 2]
```diff
@@ -2,8 +2,21 @@ import React from 'react';
 import { commonText } from '../../localization/common';
 import { useBooleanState } from '../../hooks/useBooleanState';
 import { Portal } from '../Molecules/Portal';
+import { Dashboard } from '../Dashboard';
 import { CurrentViewContext } from '../Core/CurrentViewContext';
 
+// Allowing for the class to me overriden from here
```
Comment (maxxxxxdlp) [inline src/src/components/App/index.tsx:8]: a typo: `to me overriden`
should be `to be overriden`

File: src/src/components/Dashboard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { Widget } from '../Widget';
+
+export const Dashboard = (props: { closeHandler: () => void; widgets: Array<WidgetObj> }) => {
+    
+    // NOTE: We might need to change the spacing a bit to center header better
+    return (
+        <> 
+            <div className='flex flex-row' >
+                
+                <h1 className='mx-auto my-0 text-center'>DASHBOARD</h1>
```
Comment (maxxxxxdlp) [inline src/src/components/Dashboard/index.tsx:12]: [non-blocking]

The `DASHBOARD` text is probably just temporary at this point, but for the, could you please add any text strings into the `common.tsx` file in this directory - https://github.com/maxxxxxdlp/calendar-stats/tree/main/src/src/localization

Then, you can call it in code like `commonText('yourKey')`

You can also provide optional attributes to that function (like `commonText('key', 1, 2, 3)`).

See the [`README.md`](https://github.com/maxxxxxdlp/calendar-stats/tree/main/src/src/localization#readme) for more documentation

File: src/src/components/Molecules/Portal.tsx [hunk 4]
```diff
@@ -18,7 +18,11 @@ export function Portal({
 }: {
   readonly children: JSX.Element;
 }): JSX.Element {
-  const element = React.useMemo(() => document.createElement('div'), []);
+  const element = React.useMemo(() => {
+    let element = document.createElement('div');
+    element.className = "h-full"
```
Comment (maxxxxxdlp) [inline src/src/components/Molecules/Portal.tsx:23]: [non-blocking]

Could you enable the `prettier` plugin in your IDE to keep the code formatting consistent?

**Review by maxxxxxdlp — CHANGES_REQUESTED** [review]

**Review by maxxxxxdlp — APPROVED** [review]

**Pull request merged**